sourcerer-dbstore/src/main/java/org/elder/sourcerer2/dbstore/DbstoreRepositoryVersion.kt

```diff
@@ -0,0 +1,52 @@
+package org.elder.sourcerer2.dbstore
+
+import org.elder.sourcerer2.RepositoryVersion
+import org.elder.sourcerer2.StreamId
+import org.elder.sourcerer2.StreamVersion
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeFormatterBuilder
+
+data class DbstoreRepositoryVersion(
+        val timestamp: Instant,
+        val streamId: StreamId,
+        val transactionSequenceNr: Int
+) {
+    companion object {
+        val stringFormat = Regex("""\d{14}\.\d{9}:(.+):\d{4}""")
+        val dateFormat: DateTimeFormatter = DateTimeFormatterBuilder()
+                .appendPattern("yyyyMMddHHmmss.nnnnnnnnn")
+                .toFormatter()
+    }
+}
+
+fun RepositoryVersion.toDbstoreRepositoryVersion(): DbstoreRepositoryVersion {
+    // TODO: This format is very verbose and could be easily converted to a shorter one by using
+    // a binary representation and a hex/base32/base64 encoding. Keep it human readable to start as
+    // it will make it meaningful for a human at a glance.
+    if (!DbstoreRepositoryVersion.stringFormat.matches(this.version)) {
+        throw IllegalArgumentException("Version is not in expected format")
+    }
+
+    val timestampStr = version.substring(0, 24)
+    val timestamp = LocalDateTime
+            .parse(timestampStr, DbstoreRepositoryVersion.dateFormat)
+            .toInstant(ZoneOffset.UTC)
+    val streamId = StreamId.ofString(version.substring(25, version.length - 4 - 1))
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.elder.org/coding_rules#rule_key=detekt-kotlin%3AMagicNumber)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Report magic numbers. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is. It's better to declare such numbers as constants and give them a proper name. By default, -1, 0, 1, and 2 are not considered to be magic numbers. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.elder.org/coding_rules#rule_key=detekt-kotlin%3AMagicNumber)